packages/context/src/interception-proxy.ts

```diff
@@ -24,7 +24,7 @@ export type AsValueOrPromise<T> = T extends Promise<unknown>
 export type AsInterceptedFunction<T> = T extends (
```

Comment:
I wonder if the following is better definition when defining ```AsInterceptedFunction```
```ts
export type AsInterceptedFunction<T extends new (...args: InvocationArgs)> = T extends (

```

Reply:
Actually no. T is method of the service interface, not the constructor.